File: backend/hack.py
import hashlib

known_hash = '69c459dd76c6198f72f0c20ddd3c9447'

with open('/usr/share/dict/words') as words:
    for line in words:
        word = line.lstrip()
        hash = hashlib.md5(word.encode('ascii')).hexdigest()
        print(hash)
        if hash == known_hash:
            print(word)

File: backend/hack.js
const readline = require('readline')
const crypto = require('crypto');
const fs = require('fs');

const SALT = 'blah blah blah i love salty food';

const hashify = (plaintext) => 
    crypto.createHash('md5').update(plaintext).digest('hex');

const rl = readline.createInterface({
    input: fs.createReadStream('/usr/share/dict/words'),
    output: process.stdout,
    terminal: false
});

fs.open('rainbow.txt', 'w', function (err, f) {
    rl.on('line', word => {
        fs.writeFile(f, `${hashify(word + SALT)} ${word}\n`, err => {
            if (err) console.log(err);
        })
    })
});


// for (const line of readFileSync("/usr/share/dict/words").toString().split('\n')) {
//     const word = line.trim();
//     if (hashify(word) === known_hash) {
//         console.log(word);
//     }
// }

